buildenv/jenkins/common/variables-functions.groovy

```diff
@@ -616,9 +616,21 @@ def set_build_variables() {
 }
 
 def set_sdk_variables() {
+    set_sdk_versions()
     DATESTAMP = get_date()
     SDK_FILE_EXT = SPEC.contains('zos') ? '.pax' : '.tar.gz'
-    SDK_FILENAME =  "OpenJ9-JDK${SDK_VERSION}-${SPEC}-${DATESTAMP}${SDK_FILE_EXT}"
+    SDK_FILENAME = get_value(VARIABLES.misc.sdk_filename_template, BUILD_IDENTIFIER) ?: get_value(VARIABLES.misc.sdk_filename_template, 'default')
+    echo "SDK_FILENAME (before processing):'${SDK_FILENAME}'"
+    // If filename has unresolved variables at this point, use Groovy Template Binding engine to resolve them
+    if (SDK_FILENAME.contains('$')) {
+        set_sdk_impl()
+        // Add all variables that could be used in a template
+        def binding = ["SDK_IMPL":SDK_IMPL, "SPEC":SPEC, "SDK_VERSION":SDK_VERSION, "FULL_SDK_VERSION":FULL_SDK_VERSION, "DATESTAMP":DATESTAMP, "SDK_FILE_EXT":SDK_FILE_EXT]
+        def engine = new groovy.text.SimpleTemplateEngine()
+        SDK_FILENAME = engine.createTemplate(SDK_FILENAME).make(binding)
+    }
+    echo "SDK_FILENAME:'${SDK_FILENAME}'"
+
     TEST_FILENAME = "test-images.tar.gz"
```

Comment:
All the other "support" artifacts have a timestamp, likely a good idea to add that to the TEST_FILENAME as well.

Reply:
I think this is outside of the scope of this change due to it being a cross repo dependent change
https://github.com/AdoptOpenJDK/openjdk-tests/blob/2e44aa3a6d76d549c9aa5c7a249fafb1c714af73/get.sh#L187

Reply:
While its a different repo, a change will need to go into openjdk-tests at the same time to keep PR and builds functional.